src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs

```diff
@@ -397,23 +398,104 @@ private AnalyzerConfigDocument FindOrGenerateEditorConfig()
+                                // Thus, we want to keep track of whether there is an existing header that only contains [*.cs] or only
+                                // [*.vb], depending on the language.
+                                // We also keep track of the last valid header for the language.
+                                if (!SuppressionHelpers.IsCompilerDiagnostic(_diagnostic) && _isPerLanguage &&
+                                    (_language.Equals(LanguageNames.CSharp) || _language.Equals(LanguageNames.VisualBasic)))
+                                {
+                                    if (brokenUpFileExtensions.Contains("cs") && brokenUpFileExtensions.Contains("vb"))
+                                    {
+                                        lastValidSpecificHeader = mostRecentHeader;
+                                    }
+                                }
+                                else if ((SuppressionHelpers.IsCompilerDiagnostic(_diagnostic) || !_isPerLanguage) &&
+                                         brokenUpFileExtensions.Length == 1)
+                                {
+                                    if (_language.Equals(LanguageNames.CSharp) && brokenUpFileExtensions.Contains("cs"))
+                                    {
+                                        lastValidSpecificHeader = mostRecentHeader;
+                                    }
+                                    else if (_language.Equals(LanguageNames.VisualBasic) && brokenUpFileExtensions.Contains("vb"))
+                                    {
+                                        lastValidSpecificHeader = mostRecentHeader;
+                                    }
+                                }
+                                lastValidHeader = mostRecentHeader;
+                            }
+                        }
+                    }
                 }
 
-                if (mostRecentHeader != null)
+                // We want to keep track of how far this (valid) section spans.
+                if (mostRecentHeader != null &&
+                    lastValidHeader != null &&
+                    mostRecentHeader.Equals(lastValidHeader))
                 {
-                    headers[mostRecentHeader] = curLine;
+                    lastValidHeaderSpanEnd = curLine;
+                    if (mostRecentHeader.Equals(lastValidSpecificHeader))
+                    {
+                        lastValidSpecificHeaderSpanEnd = curLine;
+                    }
                 }
             }
 
-            return null;
+            // We return only the last text change in case of duplicate entries for the same rule.
+            if (!textChange.Equals(new TextChange()))
+            {
+                return new Tuple<SourceText, TextLine?, TextLine?>(result.WithChanges(textChange), lastValidHeaderSpanEnd, lastValidSpecificHeaderSpanEnd);
```

Comment:
You can write this as `return (result.WithChanges(textChange), lastValidHeaderSpanEnd, lastValidSpecificHeaderSpanEnd);`, which implicitly creates the Tuple for you.